india_compliance/gst_india/utils/transaction_data.py

```diff
@@ -201,8 +201,25 @@ def validate_transaction(self):
     def validate_non_gst_items(self):
         validate_non_gst_items(self.doc)
 
+    def group_same_items(self, unique_items, item_details, item_row):
+        unique_item_details = unique_items.setdefault(
+            (item_row.item_code, item_row.gst_hsn_code, item_details.uom),
```

Comment:
Group just based on Item code.

Throw error (validate) if HSN or UOM is different.
- Frontend and Backend